api/util.py

```diff
@@ -80,6 +83,17 @@ def commit_file(dbc, _id, datainfo, filepath, data_path, force=False):
     """Insert a file as an attachment or as a file."""
     filename = os.path.basename(filepath)
     fileinfo = datainfo['fileinfo']
+    log_path = os.path.join(os.path.split(data_path)[0], 'logs')
+    # XXX Eventually it would be good to find a more principled/constant
+    # way to do this
+    if os.path.isdir(log_path):
+        hdlr = logging.FileHandler(os.path.join(log_path, 'commit_file.log'))
+        fmt = logging.Formatter(fmt=log_fmt, datefmt=datefmt)
+        hdlr.setFormatter(fmt)
+        hdlr.setLevel(logging.DEBUG)
+        log.addHandler(hdlr)
```

Comment:
This is the thing that made debugging much easier, namely logging the `commit_file` calls to `persistent/logs/commit_file.log`. It's a bit hackishly coded right now, I suspect it will be easier to get this right with a single-repo setup.